### ISSUE TITLE
Getting "typescript-eslint/unbound-method" using a simple Svelte "on:event" syntax. A false positive?

### ISSUE THREAD

**Issue by frederikhors** [sveltejs/eslint-plugin-svelte3#103]
I'm converting to Typescript this code:

```svelte
<div
  class={"myClass"}
  style={getStyle(someVar)}
  on:animationend={myStore.action}
>
```

on the line `on:animationend={myStore.action}` typescript-eslint is complaining about:

[`Avoid referencing unbound methods which may cause unintentional scoping of `this`. eslint@typescript-eslint/unbound-method`](https://github.com/typescript-eslint/typescript-eslint/blob/v4.17.0/packages/eslint-plugin/docs/rules/unbound-method.md).

What should I do?

Change it to the below? But I like Svelte because of its laconicity!

```svelte
<!-- I don't like this -->
<div
  class={"myClass"}
  style={getStyle(someVar)}
  on:animationend={(e)=>myStore.action(e)}
>
```

Can you help me understand, please?

**Comment by JounQin** [comment]
What is `myStore.action`? It should just as same as https://github.com/sveltejs/eslint-plugin-svelte3/issues/102.

**Comment by Conduitry** [comment]
The rule sounds like it's doing what it's supposed to do. You either need to disable it, or otherwise appease the type checker. This doesn't have anything to do with Svelte.

**Comment by frederikhors** [comment]
@JounQin `myStore.action` is:

```ts
action(event: AnimationEvent | string) {
  // handle event here
},
```

@Conduitry some hours ago I found something that had to be "correct" in Svelte types. I don't think I'm asking for much but an indication, among other things useful for all those who will look after me.

**Comment by JounQin** [comment]
@frederikhors 

Use following like https://github.com/sveltejs/svelte/pull/6094

```js
action(this:void, event: AnimationEvent | string) {
  // handle event here
},
```

Or arrow function

```js
action: (event: AnimationEvent | string) => {
  // handle event here
},
``

**Comment by frederikhors** [comment]
@JounQin it works! I love you! ❤️❤️❤️

@Conduitry Simply this.

**Comment by JounQin** [comment]
I think you should ready more carefully about the rule itself before raising issues, bro.

https://github.com/typescript-eslint/typescript-eslint/blob/v4.17.0/packages/eslint-plugin/docs/rules/unbound-method.md#rule-details
